### ISSUE TITLE
Warning messages while using cansim package to download 36-10-0107-01

### ISSUE THREAD

**Issue by PhilSmith26** [mountainMath/cansim#101]
While trying to download the Statistics Canada table 36-10-0107-01 using the cansim package, I got the following warning messages. 

1: One or more parsing issues, see `problems()` for details 
2: In value[[3L]](cond) :
  Encountered a warning when converting field Estimates to a factor, please flag this as an issue in the {cansim} repository at https://github.com/mountainMath/cansim/issues.

**Comment by mountainMath** [comment]
Thanks Philip,

I think that's a problem with the StatCan table. The "Estimates" field has inconsistent values, in some cases they come with the classification code in brackets, in some cases they don't.

```
d<-get_cansim("36-10-0107") %>% 
  filter(Prices=="Chained (2012) dollars",
          `Seasonal adjustment`=="Seasonally adjusted at annual rates",
          grepl("Food and non-alcoholic beverages",Estimates)) %>% 
  filter(REF_DATE %in% c("1974-01","1974-04","1974-07")) %>% 
  select(REF_DATE,Estimates)
```
gives
```
> d
# A tibble: 3 × 2
  REF_DATE Estimates                             
  <chr>    <chr>                                 
1 1974-01  Food and non-alcoholic beverages [C11]
2 1974-04  Food and non-alcoholic beverages      
3 1974-07  Food and non-alcoholic beverages [C11]  
```

I am now looking for internally inconsistent categories and don't try to convert to factor in cases like this. The package will emit a warning message if it detects such problems, as this kind of inconsistency can lead to problems downstream. 

Please download the current release candidate for version v0.3.10 via
```
remotes::install_github("mountainmath/cansim#100")
```
to confirm that this fix works.

**Comment by PhilSmith26** [comment]
Hi Jens. Thanks for looking into this case.

The inconsistency you found for Food and non-alcoholic beverages in 
74-04 and 75-04 actually applies to all of the Estimates categories in 
those two particular quarters. It seems they dropped the hierarchical 
categories in square brackets for all of the rows for 74-04 and 75-04, 
but in no other rows for other quarters. This must just be a mistake by 
Statistics Canada. But I cannot imagine how that kind of mistake would 
be made.

Parenthetically, Statistics Canada overhauled this table in 2018 Q3. 
They introduced at least one new 'Estimates' category, Cannabis[C123], 
to allow for the fact that this substance was legalized in October 2018. 
They carried those cannabis estimates back to 1961 though, based on 
estimates of illegal cannabis sales, so there is no inconsistency there. 
There could have been, though, if they had introduced cannabis as a new 
product in 2018 and did not carry it back to 1961.

The more significant change from a structural point of view relates to 
the 'Prices' category. They had been scaling the 'Chained dollars' 
estimates to 2007 dollars. Then quite a while back they began scaling 
them to 2012 dollars, but they kept the 2007 version as well. Then as of 
2018 Q3 they dropped the 2007 version going forward and now show only 
the 2012 version. So one can restore the structural integrity of the 
table in this respect by filtering with Prices!="Chained (2007) 
dollars".

This kind of change to a table by Statistics Canada is, I believe, quite 
rare. What they usually do is terminate the table entirely and start a 
new one. Users are not keen on this approach, however, because it breaks 
the time series. Users have to retrieve two, or more, tables and then 
splice the data together themselves if they want a longer time series, 
which is annoying. In this table, there is continuity from 1961 to today 
for the 'Estimates' categories. So if, for example, I want to retrieve 
data for Food [C111], seasonally adjusted at annual rates, I can easily 
get the entire time series from 1961 Q1 to 2021 Q2 using vector 
v61989019. This is wonderful if I am just interested in a few time 
series, but if I want to use the entire table it gets awkward because of 
the structural changes. I only get into problems if I want the 
Prices!="Chained (2007) dollars" data, since this is available only from 
1961 to 2018 Q2.

So if done properly by Statistics Canada, structural changes to some of 
the tables (rather than terminating and starting anew) can be a very 
good thing. Perhaps the inconsistency warning issued by the cansim 
package should say that this could be either a problem with the data or 
an intentional structural change to the table by Statistics Canada.

If Statistics Canada would just correct the mistakes in 74-04 and 75-04 
and also delete the Prices!="Chained (2007) dollars" data, which is 
useless anyway, the table would be okay. I'm going to contact them about 
that.

Philip

**Comment by mountainMath** [comment]
By "inconsistent" I mean that the metadata is inconsistent with the data in the table. For Prices, the metadata has both the "Chained (2012) dollars" and "Chained (2007) dollars" (as well as "Current prices"  and "Contributions to percent change"), but for the Estimates the metadata only has the values without the classification codes appended. This happens for some other tables too and the {cansim} allows for table fields having the classification codes appended.

I suspect that StatCan does that because fields can have duplicated names, which might cause problems for naive users of the data. The {cansim} package now automatically de-duplicates duplicate field names in order to safeguard against downstream problems. StatCan seems to append the classification code for some tables where there are duplicate field names, but not in all of them. In this particular case, StatCan did not append the classification codes consistently. Here is the output for fields containing the word "Communication" for the same years as above.
```
# A tibble: 6 × 3
  REF_DATE Estimates             `Classification Code for Estimates`
  <chr>    <chr>                 <chr>                              
1 1974-01  Communications [C18]  [C18]                              
2 1974-01  Communications [C181] [C181]                             
3 1974-04  Communications        [C18]                              
4 1974-04  Communications        [C181]                             
5 1974-07  Communications [C18]  [C18]                              
6 1974-07  Communications [C181] [C181]   
```

One of us should probably notify StatCan of this problem.

**Comment by PhilSmith26** [comment]
I have raised the issue with Statistics Canada and I'll let you know the 
outcome.

Philip

**Comment by mountainMath** [comment]
Workaround implemented in v0.3.10
